orchestrator/schedules/validate_subscriptions.py

```diff
@@ -34,29 +37,19 @@
 def validate_subscriptions() -> None:
     if app_settings.VALIDATE_OUT_OF_SYNC_SUBSCRIPTIONS:
         # Automatically re-validate out-of-sync subscriptions. This is not recommended for production.
-        select_query = select(SubscriptionTable).join(ProductTable)
+        subscriptions = get_subscriptions_on_product_table()
     else:
-        select_query = select(SubscriptionTable).join(ProductTable).filter(SubscriptionTable.insync.is_(True))
-    subscriptions = db.session.scalars(select_query)
-    for subscription in subscriptions:
-        validation_workflow = None
-
-        for workflow in subscription.product.workflows:
-            if workflow.target == Target.SYSTEM:
-                validation_workflow = workflow.name
-
-        if validation_workflow:
-            default = TARGET_DEFAULT_USABLE_MAP[Target.SYSTEM]
-            usable_when = WF_USABLE_MAP.get(validation_workflow, default)
+        subscriptions = get_subscriptions_on_product_table_in_sync()
 
-            if subscription.status in usable_when:
-                json = [{"subscription_id": str(subscription.subscription_id)}]
+    for subscription in subscriptions:
+        system_product_workflows = get_system_product_workflows_for_subscription(subscription)
 
-                validate_func = get_execution_context()["validate"]
-                validate_func(validation_workflow, json=json)
-        else:
+        if not system_product_workflows:
             logger.warning(
                 "SubscriptionTable has no validation workflow",
                 subscription=subscription,
                 product=subscription.product.name,
             )
+            break
```

Comment:
I believe this should be `continue` to loop through the next iter instead of breaking the entire loop
```suggestion
            continue
```

Reply:
TY, nice catch. should be continue indeed